l1-contracts/contracts/bridge/asset-router/AssetRouterBase.sol

```diff
@@ -0,0 +1,368 @@
+            if (block.chainid != L1_CHAIN_ID) {
+                /// @dev Legacy bridging is only supported for L1->L2 native token transfers.
+                revert InvalidChainId();
+            }
+            (assetId, transferData) = _handleLegacyData(_data, _prevMsgSender);
+        } else {
+            revert UnsupportedEncodingVersion();
+        }
+
+        if (BRIDGE_HUB.baseTokenAssetId(_chainId) == assetId) {
+            revert AssetIdNotSupported(assetId);
+        }
+
+        bytes memory bridgeMintCalldata = _burn({
+            _chainId: _chainId,
+            _msgValue: _value,
+            _assetId: assetId,
+            _prevMsgSender: _prevMsgSender,
+            _transferData: transferData,
+            _passValue: true
+        });
+
+        bytes32 txDataHash = _encodeTxDataHash(encodingVersion, _prevMsgSender, assetId, transferData);
+        request = _requestToBridge({
+            _prevMsgSender: _prevMsgSender,
+            _assetId: assetId,
+            _bridgeMintCalldata: bridgeMintCalldata,
+            _txDataHash: txDataHash
+        });
+
+        emit BridgehubDepositInitiated({
+            chainId: _chainId,
+            txDataHash: txDataHash,
+            from: _prevMsgSender,
+            assetId: assetId,
+            bridgeMintCalldata: bridgeMintCalldata
+        });
+    }
+
+    /// @inheritdoc IAssetRouterBase
+    function bridgehubConfirmL2Transaction(
+        uint256 _chainId,
+        bytes32 _txDataHash,
+        bytes32 _txHash
+    ) external virtual override {}
+
+    /// @inheritdoc IAssetRouterBase
+    function getDepositCalldata(
+        address _sender,
+        bytes32 _assetId,
+        bytes memory _assetData
+    ) public view override returns (bytes memory) {
+        // First branch covers the case when asset is not registered with NTV (custom asset handler)
+        // Second branch handles tokens registered with NTV and uses legacy calldata encoding
+        if (nativeTokenVault.tokenAddress(_assetId) == address(0)) {
+            return abi.encodeCall(IL1AssetRouter.finalizeDeposit, (block.chainid, _assetId, _assetData));
+        } else {
+            // slither-disable-next-line unused-return
+            (, address _receiver, address _parsedNativeToken, uint256 _amount, bytes memory _gettersData) = DataEncoding
+                .decodeBridgeMintData(_assetData);
+            return
+                _getLegacyNTVCalldata({
+                    _sender: _sender,
+                    _receiver: _receiver,
+                    _parsedNativeToken: _parsedNativeToken,
+                    _amount: _amount,
+                    _gettersData: _gettersData
+                });
+        }
+    }
+
+    function _getLegacyNTVCalldata(
+        address _sender,
+        address _receiver,
+        address _parsedNativeToken,
+        uint256 _amount,
+        bytes memory _gettersData
+    ) internal view virtual returns (bytes memory) {}
```

Comment:
```suggestion
    ) internal view virtual returns (bytes memory);
```

Same here, if a function should have an implementation it is better to not forget to put it